test/docs/sonic-dash-test-HLD.md

```diff
@@ -0,0 +1,212 @@
+Although it is intended that the API endpoints are running natively on the DUT itself, it is conceivable that a software agent/proxy can provide this endpoint, e.g. in an appliance which hosts a SmartNIC adaptor card. Such an agent *could* translate the standardized endpoint API calls into whatever form the DUT requires, e.g. PCIe bus driver calls directly to the device, or into a non-standard proprietary RPC which then gets relayed to the device.  In all cases, the Test Controller should not have to treat the combination of proxy + device any differently than it treats an integrated device with onboard API endpoint. See Figure 6.
+
+**Figure 6. Out-of-band Access to DUT API Endpoint Via Proxy**
+
+![dash-testbed-inband-api.](../images/dash-testbed-out-of-band-via-proxy.svg)
+
+# Reference Testbed Architecture - System-Level & Multiple DUTs
+**TODO**
+
+# Standardized DUT Configuration APIs
+All DUTs will provide the following socket-based interfaces for configuring and controlling the devices. These are device service endpoints which are accessed via software clients running in the Controller.
+
+Vendors *should* provide a convenient, automatable means to select between a production mode using [gNMI](https://https://github.com/openconfig/gnmi), or a dataplane evaluation image using [saithrift](https://github.com/opencomputeproject/SAI/tree/master/test/saithrift). Vendor/SKU-specific mechanisms should be included in the `/targets` directory and accessible via the automation software at test run time. As an alternative, DUTS can be supplied with fixed images to operate in one mode or the other, as long as a means to switch between these images (mode switch, image reload, etc.) is provided.
+
+> **NOTE**: At this time, the SONiC DASH project is nascent and the early emphasis will be on performance and conformance of the underlying dataplane. Therefore, the DASH-SAI interface will be the early focus in order to eliminate the complications of debugging problems involving the entire SONiC stack. The [saithrift](https://github.com/opencomputeproject/SAI/tree/master/test/saithrift) server approach will be used for initial "lab" testing. By definition, it is not a production interface. Later on, the gNMI interface will be used for production and dataplane testing.
+
+## gNMI Northbound API
+This is the canonical SONiC appliance management interface and it conforms to the standard SONiC Management requirements, see [SONiC Management Framework](https://github.com/Azure/SONiC/blob/master/doc/mgmt/Management%20Framework.md#3123-gnmi). This uses [gNMI](https://https://github.com/openconfig/gnmi) with [OpenConfig](https://github.com/openconfig) data models. All SONiC-DASH devices are expected to eventually conform to and be tested over this interface. See Figure 7. (Note, this diagram was adapted from the standard [SONiC Architecture](https://github.com/Azure/SONiC/wiki/Architecture) documents and may contain elements not-applicable to SONIC-DASH DUTs).
```

Comment:
gnmi to sai will be platform independent, therefore they will be covered mostly in the sonic unit test and integration test, the dash dataplane test should be covered mostly using the thrift api dataplane test. This allows the troubleshoot easy and isolate the problem.

Reply:
Sure, I understand. Would you like me to add that clarifying language to this spec?

Reply:
Hi Gerald, thanks for the feedback. Is that comment directed at me, and is there something this spec should say with regard to that? We'll surely get more specific as we develop the test cases themselves.